storage/api.go

```diff
@@ -272,4 +272,15 @@ type TargetStorage interface {
 
 	// Wipe removes all contents of the target storage.
 	Wipe(ctx context.Context) error
+
+	// DisableTriggersAndFKConstraints disables all triggers and foreign key constraints
+	// in indexer tables. This is useful when inserting blockchain data out of order,
+	// so that later blocks can refer to (yet unindexed) earlier blocks without violating constraints.
+	DisableTriggersAndFKConstraints(ctx context.Context) error
+
+	// EnableTriggersAndFKConstraints enables all triggers and foreign key constraints
+	// in the given schema.
+	// WARNING: This might enable triggers not explicitly disabled by DisableTriggersAndFKConstraints.
+	// WARNING: This does not enforce/check contraints on rows that were inserted while triggers were disabled.
```

Comment:
Would it be worthwhile to eventually validate the constraints on the inserted data, at some point after the parallel analyzers have all finished? 

It seems like it's not trivial, we'd have to delete all the constraints, then re-add them which would be expensive. Plus, even enumerating all the constarints for a single table is a [bit gnarly](https://dba.stackexchange.com/questions/214863/how-to-list-all-constraints-of-a-table-in-postgresql). But it would be a useful sanity check to have, even as a one-off similar to the statecheck

Reply:
Agreed that it would be nice to have, and agreed that it's not trivial. 
Opened https://app.clickup.com/t/866a7qg4j

Reply:
@mitjat:

> The current fast-sync solution (still not fully implemented, so things might change, but unlikely) does not actually disable FKs, so this task is not needed.